src/resolvers/Mutation/removeComment.ts

```diff
@@ -48,38 +52,42 @@ export const removeComment: MutationResolvers["removeComment"] = async (
     );
   }
 
-  // Checks whether currentUser with _id === context.userId is not the creator of comment.
-  if (comment.creator.toString() !== context.userId.toString()) {
+  const isCurrentUserAdminOfOrganization = currentUser.adminFor.some(
+    (organization) => organization.equals(comment.postId.organization)
+  );
+
+  // Checks whether currentUser with _id === context.userId has the authorization to delete the comment
+  if (
+    currentUser.userType !== "SUPERADMIN" &&
```

Comment:
These should be `||` OR conditions not `&&` AND.

Reply:
Lets suppose the user is a SUPERADMIN. Then the conditional in the first line would evaluate to False, and we would break out of the if clause and thus wouldn't throw an error. If we were to use OR, then we would proceed to check for other conditions too. 

The same logic can be extended to the other conditions as well. We are using AND here as the error should only be thrown if all three conditions are evaluated to true, i.e. the user does not have any of the three listed authorizations.